sys/net/gnrc/network_layer/ndp/gnrc_ndp.c

```diff
@@ -331,8 +331,10 @@ void gnrc_ndp_nbr_adv_send(const ipv6_addr_t *tgt, gnrc_netif_t *netif,
     gnrc_netif_acquire(netif);
     do {    /* XXX: hidden goto */
         int tgt_idx;
+        gnrc_netif_t *tgt_netif = gnrc_netif_get_by_ipv6_addr(tgt);
 
-        if ((tgt_idx = gnrc_netif_ipv6_addr_idx(netif, tgt)) < 0) {
+        if ((tgt_netif == NULL) ||
+            (tgt_idx = gnrc_netif_ipv6_addr_idx(tgt_netif, tgt)) < 0) {
```

Comment:
This seems redundant first you get the interface by the address, then you check if the interface is assigned to that interface. Trying to cover race conditions should not be necessary due to the `gnrc_netif_acquire(netif)` above, so I propose to make the checking of `tgt_idx` an assert after the if failed:

```c
if (tgt_netif == NULL) {
    DEBUG("ndp: tgt not assigned to interface. Abort sending\n");
    break;
}
tgt_idx = gnrc_netif_ipv6_addr_idx(tgt_netif, tgt);
assert(tgt_idx >= 0);
```

Reply:
applied